rudder-core/src/main/scala/com/normation/cfclerk/domain/TechniqueTemplate.scala

```diff
@@ -40,18 +40,21 @@ import com.normation.utils.HashcodeCaching
 
 
 /**
- * Representation of a template qualified name.
- * A template is linked to a policy, so that two
- * different policies can have templates with identical
- * names.
+ * Representation of a technique resource id.
  *
- * A template qualified name has a string representation
- * that is by convention "policy name" / "template name".
+ * A resource may be either defined relatively to a technique, or relatively to
+ * the configuration-repository directory (the git root directory).
  */
-case class Cf3PromisesFileTemplateId(val techniqueId:TechniqueId, val name:String) {
-  override def toString() = techniqueId.toString + "/" + name
+sealed trait TechniqueResourceId {
+  def name: String
 }
 
+//the one relative to the technique
+final case class TechniqueResourceIdByName(techniqueId: TechniqueId, name: String) extends TechniqueResourceId
+
+//the one relative to configuration-repository - actually, that resource is not really link to a technique
+final case class TechniqueResourceIdByPath(parentDirectories: List[String], name: String) extends TechniqueResourceId
```

Comment:
you should explain why it takes a list of parent directory, and that if the list is empty, the behaviour is different